pkg/controllers/dynakube/otelc/service/reconciler.go

```diff
@@ -42,37 +42,51 @@ func NewReconciler(client client.Client, apiReader client.Reader, dk *dynakube.D
 
 func (r *Reconciler) Reconcile(ctx context.Context) error {
 	if !r.dk.TelemetryIngest().IsEnabled() {
-		return r.removeServiceOnce(ctx)
-	}
+		r.removeServiceOnce(ctx)
 
-	if r.dk.TelemetryIngest().ServiceName != "" {
-		return r.removeServiceOnce(ctx)
+		return nil
 	}
 
+	r.removeServices(ctx, r.dk.TelemetryIngest().GetServiceName())
+
 	return r.createOrUpdateService(ctx)
 }
 
-func (r *Reconciler) removeServiceOnce(ctx context.Context) error {
+func (r *Reconciler) removeServiceOnce(ctx context.Context) {
 	if meta.FindStatusCondition(*r.dk.Conditions(), serviceConditionType) == nil {
-		return nil
+		return
 	}
 	defer meta.RemoveStatusCondition(r.dk.Conditions(), serviceConditionType)
 
-	svc, err := r.buildService()
-	if err != nil {
-		log.Error(err, "could not build service during cleanup")
+	r.removeServices(ctx, "")
+}
 
-		return err
+func (r *Reconciler) removeServices(ctx context.Context, actualServiceName string) {
```

Comment:
```suggestion
func (r *Reconciler) removeServices(ctx context.Context, actualServiceName string) {
```
very confusing method name, at first I thought it's removing serviceName with actualServiceName.
then I notices it's plural.

IMO better to be more explicit what it does, `removeAllServicesExcept(ctx, actualServiceName)`